daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/CollisionSpec.scala

```diff
@@ -101,31 +106,44 @@ class CollisionSpec extends AnyWordSpec with Matchers {
           }
         """
 
-      // a package with collision: two constructs have the same fully resoled name "A.B"
-      val positiveTestCase1 = p"""
+      val positiveTestCases = Table(
+        "module",
+        // a package with collision: two constructs have the same fully resoled name "A.B"
+        p"""
           module A {                    // fully resolved name: "A"
             record B = {};              // fully resolved name: "A.B" (collision)
           }
 
           module A.B {                  // fully resolved name: "A.B" (collision)
 
           }
-        """
-
-      // a package with collision: two constructs have the same fully resoled name "A.B"
-      val positiveTestCase2 = p"""
+        """,
+        // a package with collision: two constructs have the same fully resoled name "A.B"
+        p"""
           module A {                    // fully resolved name: "A"
             record B = {};                // fully resolved name: "A.B" (collision)
           }
 
           module a.B {                  // fully resolved name: "a.B" (collision)
 
           }
-        """
+        """,
+        // a package with collision: two constructs have the same fully resoled name "A.B"
+        p"""
+          module A {                    // fully resolved name: "A"
+            synonym B = |Mod:A|;        // fully resolved name: "A.B" (collision)
+          }
+
+          module a.B {                  // fully resolved name: "a.B" (collision)
+
```

Comment:
I'm not sure that this is testing collision between module and type synonym, since this also has a collision between module `A` and (virtual) module `a`. Maybe something like this would isolate the synonym/module collision:

```
module A {
   synonym B = |Mod:A|;
   
   module b {
   }
}
```